cpp/arcticdb/processing/clause.cpp

```diff
@@ -367,6 +380,11 @@ Composite<EntityIds> AggregationClause::process(Composite<EntityIds>&& entity_id
     // Work out the common type between the processing units for the columns being aggregated
     procs.broadcast([&aggregators_data, &aggregators=aggregators_](auto& proc) {
```

Comment:
I think this works kind of by luck, in that `broadcast` iterates forward through the contained vector in the composite, which is in order because of the way `procs` was constructed. However, this is an implementation detail of `Composite`, and should not be relied upon. Replacing `procs.broadcast` with `for (auto& proc: procs_as_range)` is guaranteed to have the correct behaviour.